lib/hexo/router.js

```diff
@@ -14,11 +14,28 @@ class RouteStream extends Readable {
     this.modified = data.modified;
   }
 
+  // Assume we only accept Buffer, plain object, or string
+  _toBuffer(data) {
+    if (data instanceof Buffer) {
+      return data;
+    }
+    if (typeof data === 'object') {
+      data = JSON.stringify(data);
```

Comment:
comment: is Buffer conversion also needed here?

Reply:
It will be handled by L25 then.

Reply:
This is to allow user to pass plain objects directly to the route without the need of stringify-ing them or wrap them with a function. Before this change, you would have to use
```js
hexo.extend.generator.register("resource", () => [
  {
    path: "data.json",
    data: JSON.stringify({
      // some_plain_object
    }),
  },
]);
```
or
```js
hexo.extend.generator.register("resource", () => [
  {
    path: "data.json",
    data: () => ({
      // some_plain_object
    }),
  },
]);
```

after:
```js
hexo.extend.generator.register("resource", () => [
  {
    path: "data.json",
    data: {
      // some_plain_object
    },
  },
]);
```

This change makes the data type of `data` more consistent. It should have no impact on existing generators since setting a plain object directly in `data` will result an error right now.

Reply:
> It will be handled by L25 then.

I misread that line as `else if`.